modtree/modtree
# feat: split docker build steps

### Summary of changes

- There are now 4 images up on the [public docker registry](https://hub.docker.com/u/modtree)
- What each does is covered at [`/docker/README.md`](https://github.com/modtree/modtree/blob/4bafb518f7f538b49e34c8e215c9cd5262283840/docker/README.md)
- Goal: make rebuilding `modtree/server` image much faster by having all yarn packages already installed.

### Testing

- Go to `/docker`
- `docker compose up -d`
- Go to `localhost:8080`. It should say that the modtree server is running.
- `nx serve web`
- Searchbar at `localhost:3000` should work.

### Deeper testing
- console.log something in `apps/server/src/main.ts`
- go to `/docker` again
- rebuild the server container image with `./build Dockerfile.server`
- see your build output with `docker images`
  - it should have `REPOSITORY` as `modtree/server` and `TAG` as `latest`.
  - It should also have a unique `IMAGE ID`.
- re-tag that image with `docker tag [IMAGE ID] modtree/server:0.5.420blaze`
- edit `docker-compose.yml`
  - change the line containing `image: 'modtree/server:0.5.1'` and update it with your new tag
- run `docker compose down` if you haven't already
- `docker compose up` without the `-d` flag. This will let you see all the logs.
- try and spot your console.log that you set in the first step.
